documentation/src/main/asciidoc/topics/anchored_keys.adoc

```diff
@@ -0,0 +1,179 @@
+
+The cache will fail to start if these attributes are explicitly set to other values,
+if state transfer is disabled, or if transactions are enabled.
+
+
+== Implementation status
+
+Basic operations are implemented: `put`, `putIfAbsent`, `get`, `replace`, `remove`, `putAll`, `getAll`.
+
+
+=== Functional commands
+The `FunctionalMap` API is not implemented.
+
+Other operations that rely on the functional API's implementation do not work either: `merge`, `compute`,
+`computeIfPresent`, `computeIfAbsent`.
+
+=== Partition handling
+When a node crashes, surviving nodes do not remove anchor references pointing to that node.
+In theory, this could allow merges to skip conflict resolution, but currently the `PREFERRED_NON_NULL`
+merge policy is configured automatically and cannot be changed.
+
+=== Listeners
+Cluster listeners and client listeners are implemented and receive the correct notifications.
+
+Non-clustered embedded listeners currently receive notifications on all the nodes, not just the node
+where the value is stored.
+
+
+== Performance considerations
+
+=== Client/Server Latency
+The client always contacts the primary owner, so any read has a
+`(N-1)/N` probability of requiring a unicast RPC from the primary to the anchor owner.
+
+Writes require the primary to send the value to one node and the anchor address
+to all the other nodes, which is currently done with `N-1` unicast RPCs.
+
+In theory we could send in parallel one unicast RPC for the value and one multicast RPC for the address,
+but that would need additional logic to ignore the address on the anchor owner
+and with TCP multicast RPCs are implemented as parallel unicasts anyway.
+
+
+=== Memory overhead
+Compared to a distributed cache with one owner, an anchored-keys cache
+contains copies of all the keys and their locations, plus the overhead of the cache itself.
+
+Therefore, a node with anchored-keys caches should stop accepting new entries when it has less than
+`(<key size> + <per-key overhead>) * <number of entries not yet inserted>` bytes available.
+
+NOTE: The number of entries not yet inserted is obviously very hard to estimate.
+In the future we may provide a way to limit the overhead of key location information,
+e.g. by using a distributed cache.
+
+The per-key overhead is lowest for off-heap storage, around 63 bytes:
+8 bytes for the entry reference in `MemoryAddressHash.memory`,
+29 bytes for the off-heap entry header,
+and 26 bytes for the serialized `RemoteMetadata` with the owner's address.
+
+The per-key overhead of the ConcurrentHashMap-based on-heap cache,
+assuming a 64-bit JVM with compressed OOPS, would be around 92 bytes:
+32 bytes for `ConcurrentHashMap.Node`, 32 bytes for `MetadataImmortalCacheEntry`,
+24 bytes for `RemoteMetadata`, and 4 bytes in the `ConcurrentHashMap.table` array.
+
+
+=== State transfer
+State transfer does not transfer the actual values, but it still needs
+to transfer all the keys and the anchor owner information.
```

Comment:
Maybe: "State transfer does not transfer values, only keys and anchor owner information.